zrml/global-disputes/src/lib.rs

```diff
@@ -243,13 +315,76 @@ mod pallet {
                 ExistenceRequirement::AllowDeath,
             )?;
 
+            Self::update_winner(&market_id, &outcome, voting_outcome_fee);
+            let possession =
+                Some(Possession::Paid { owner: owner.clone(), fee: voting_outcome_fee });
+            let outcome_info = OutcomeInfo { outcome_sum: voting_outcome_fee, possession };
+            <Outcomes<T>>::insert(market_id, outcome.clone(), outcome_info);
+
             Self::deposit_event(Event::AddedVotingOutcome { market_id, owner, outcome });
             // charge weight for successfully have no owners in Winners
             // this is the case, because owners are not inserted
             // as long as the global dispute is not finished
             Ok((Some(T::WeightInfo::add_vote_outcome(0u32))).into())
         }
 
+        /// Return the voting outcome fees in case the global dispute was destroyed.
+        ///
+        /// # Arguments
+        ///
+        /// - `market_id`: The id of the market.
+        ///
+        /// # Weight
+        ///
+        /// Complexity: `O(n)`,
+        /// where `n` is the number of all existing outcomes for a global dispute.
+        #[frame_support::transactional]
+        #[pallet::weight(T::WeightInfo::refund_vote_fees(
+            T::RemoveKeysLimit::get(),
+            T::MaxOwners::get()
+        ))]
+        pub fn refund_vote_fees(
+            origin: OriginFor<T>,
+            #[pallet::compact] market_id: MarketIdOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?;
+
+            let gd_info = <GlobalDisputesInfo<T>>::get(market_id)
+                .ok_or(Error::<T>::NoGlobalDisputeInitialized)?;
+            ensure!(gd_info.status == GDStatus::Destroyed, Error::<T>::UnfinishedGlobalDispute);
+
+            let mut owners_len = 0u32;
+            let mut removed_keys_amount = 0u32;
+            for (_, outcome_info) in
+                <Outcomes<T>>::drain_prefix(market_id).take(T::RemoveKeysLimit::get() as usize)
+            {
+                match outcome_info.possession {
+                    Some(Possession::Paid { owner, fee }) => {
+                        T::Currency::transfer(
+                            &Self::reward_account(&market_id),
+                            &owner,
+                            fee,
+                            ExistenceRequirement::AllowDeath,
```

Comment:
Hmm, not quite sure what I was thinking here. The problem with `AllowDeath` is that it allows the account to get dusted, which may result in a loss of funds. Is it possible for the `fee` to be less than the existential despoit of ZTG?

Reply:
The current configuration parameter `VotingOutcomeFee` is `200 ZTG`.

Reply:
Okay, then we're safe. Could you add a docstring to `VotingOutcomeFee` which says that it shouldn't be less than the existential deposit? I wonder if there's a good way to enforce this.

Reply:
You could add a debug_assert!

Reply:
In `add_vote_outcome` I switched to KeepAlive so that the vote outcome fee is at least the existential deposit. I currently can't push to github because of a server incident, but I will push later.

Reply:
97e4dcccbf730f22f6665dafa85f676f133ecd91